src/Dotnet.Script.DependencyModel.Nuget/NuGetSourceReferenceResolver.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+
+namespace Dotnet.Script.DependencyModel.NuGet
+{
+    /// <summary>
+    /// A <see cref="SourceReferenceResolver"/> decorator that handles
+    /// references to NuGet packages in scripts.  
+    /// </summary>
+    public class NuGetSourceReferenceResolver : SourceReferenceResolver
```

Comment:
in order to make this work with the REPL there needs to be some sort of a public method that allows us to mutate the script map. Something like `WithScriptMap()` or so

So when I start a REPL, original options are created without script map. But as soon as I do `#load` from nuget, the resolver should get a new map discovered based on the current `#load`(or many `#loads`). I'm just not sure if we do it multiple times in a REPL session, should we keep appending to the map or replace it every time, I think replace it every time since the earlier compilations will already have been completed

Reply:
I'll look into it. Will create a test for this in the `InteractiveRunnerTests` class

Reply:
actually `ScriptOptions` itself is immutable so it needs to be reset every time. This means that it should be enough to just do `_scriptOptions = _scriptOptions.WithSourceResolver(...)` inside the `Interactiverunner` and pass in a whole new instance of `NuGetSourceReferenceResolver` with a new source map extracted from the current REPL submission. I'm just not sure if we can safely reset the resolver like that, we should be able though, as the previous REPL submissions are already compiled in memory

Reply:
but again, this will be not needed anymore as soon as we have a good story for "self-contained" resolvers (such that don't need to modify `ScriptOptions`) which I really hope we can have at some point